### PR TITLE
Improved example and small fixes to afnumpy

### PR DESCRIPTION
**Pull request by daurer**
- Made "phase retrieval" example more usable (self-contained)
- Small fixes in afnumpy (e.g. meshgrid)
- Fixing travis builds (arrayfire has changed their build commands)
- Added "\_\_format\_\_" method to afnumpy multiarray object

### CODE REVIEW

**Review by FilipeMaia — COMMENTED** [review]
Instead of commenting it out, leave it but mark the function as `@xfail`, meaning a known failure.

File: tests/test_ndarray.py [hunk 1]
```diff
@@ -399,7 +399,8 @@ def test_getitem():
     b = numpy.ones((3))
     a = afnumpy.array(b)
     iassert(a[1:4],b[1:4])
-    iassert(a[3::-1],b[3::-1])
+    # This no longer works with current version of arrayfire, accessing a d_array with slice(None,-1,-1) does not give the expected output
+    #iassert(a[3::-1],b[3::-1])
```
Comment (FilipeMaia) [inline tests/test_ndarray.py]: My previous comment was related to this line.

Comment (daurer) [reply tests/test_ndarray.py]: done

**Comment by daurer** [comment]
This code example produces a segmentation fault on my GPU:

```python
import afnumpy as af
import afnumpy.fft as fft
a = fft.fftn(af.zeros(100))
```

can you reproduce?

**Pull request merged**